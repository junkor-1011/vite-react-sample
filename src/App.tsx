import { useState } from 'react';
import { css } from '@emotion/react';
import reactLogo from './assets/react.svg';
import './App.scss';
import Clock from '@/components/atoms/Clock';

const textBoldStyle = css({
  fontWeight: 'bold',
});

const greenStyle = css`
  color: green;
`;

const App: React.FC = () => {
  const [count, setCount] = useState(0);

  return (
    <div className="App" css={textBoldStyle} data-testid="App">
      <div>
        <a href="https://vitejs.dev" target="_blank" rel="noreferrer">
          <img src="/vite.svg" className="logo" alt="Vite logo" />
        </a>
        <a href="https://reactjs.org" target="_blank" rel="noreferrer">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>
      <div className="card">
        <button
          onClick={() => {
            setCount((count) => count + 1);
          }}
        >
          <p css={greenStyle}>count is {count}</p>
        </button>
        <p css={{ color: 'blue' }}>
          Edit <code>src/App.tsx</code> and save to test HMR
        </p>
      </div>
      <p className="read-the-docs">Click on the Vite and React logos to learn more</p>
      <div>
        <Clock css={{ color: 'blue', fontWeight: 'bold' }} />
      </div>
    </div>
  );
};

export default App;
