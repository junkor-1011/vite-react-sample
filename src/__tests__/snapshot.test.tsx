import { screen, render } from '@testing-library/react';
import { createSerializer } from '@emotion/jest';
import App from '../App';

/** 2022-09-18T08:52:10Z */
const dummyUnixtime = 1663491130; // [sec]
/** 2022-09-18T08:52:10.000Z */
const dummyTime = dummyUnixtime * 1000;

expect.addSnapshotSerializer(createSerializer());

describe('snapshot test', () => {
  beforeEach(() => {
    vitest.useFakeTimers();
    vitest.setSystemTime(dummyTime);
  });
  afterEach(() => {
    vitest.clearAllMocks();
    vitest.clearAllTimers();
    vitest.useRealTimers();
  });
  it('App Component', () => {
    render(<App />);
    expect(screen.getByTestId('App')).toMatchSnapshot();
  });
});
