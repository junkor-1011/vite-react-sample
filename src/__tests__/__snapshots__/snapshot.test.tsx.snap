// Vitest Snapshot v1

exports[`snapshot test > App Component 1`] = `
.emotion-0 {
  font-weight: bold;
}

.emotion-1 {
  color: green;
}

.emotion-2 {
  color: blue;
}

.emotion-3 {
  color: blue;
  font-weight: bold;
}

<div
  class="App emotion-0"
  data-testid="App"
>
  <div>
    <a
      href="https://vitejs.dev"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="/vite.svg"
      />
    </a>
    <a
      href="https://reactjs.org"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <button>
      <p
        class="emotion-1"
      >
        count is 
        0
      </p>
    </button>
    <p
      class="emotion-2"
    >
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
  <div>
    <span
      class="emotion-3"
    >
      2022/9/18 17:52:10
    </span>
  </div>
</div>
`;
